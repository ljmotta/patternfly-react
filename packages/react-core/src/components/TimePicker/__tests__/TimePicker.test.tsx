import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { TimePicker } from '../TimePicker'

describe('test timepicker onChange method with valid values', () => {
  const testOnChange = ({ input, expects }) => {
    const onChange = jest.fn();

    const event: any = {
      currentTarget: { value: input.value }
    };
    const view = mount(<TimePicker onChange={onChange} value={input.value} is24Hour={input.is24Hour} aria-label="time picker" />);
  
    act(() => {
      view.find('input').prop('onChange')(event);
    })
    expect(onChange).toBeCalledWith(input.value, expects.hour, expects.minutes);
  }

  test('should return the correct value using the AM/PM pattern - midnight', () => {
    testOnChange({ input: { value: "12:00 AM", is24Hour: false }, expects: { hour: 0, minutes: 0 }})
  })

  test('should return the correct value using the AM/PM pattern - midday', () => {
    testOnChange({ input: { value: "12:35 PM", is24Hour: false }, expects: { hour: 12, minutes: 35 }})
  })

  test('should return the correct value using the AM/PM pattern - last minute of the day', () => {
    testOnChange({ input: { value: "11:59 PM", is24Hour: false }, expects: { hour: 23, minutes: 59 }})
  })

  test('should return the correct value using the 24h pattern - midnight', () => {
    testOnChange({ input: { value: "00:00", is24Hour: true }, expects: { hour: 0, minutes: 0 }})
  })

  test('should return the correct value using the 24h pattern - midday', () => {
    testOnChange({ input: { value: "12:35", is24Hour: false }, expects: { hour: 12, minutes: 35 }})
  })

  test('should return the correct value using the 24h pattern - last minute of the day', () => {
    testOnChange({ input: { value: "23:59", is24Hour: true }, expects: { hour: 23, minutes: 59 }})
  })
})
