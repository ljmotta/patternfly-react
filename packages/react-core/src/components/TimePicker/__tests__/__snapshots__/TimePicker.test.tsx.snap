// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test isInvalid should be invalid after onChange 1`] = `
<TimePicker
  aria-label="time picker"
  className=""
  delimiter=":"
  direction="down"
  inputProps={Object {}}
  invalidFormatErrorMessage="Invalid time format"
  is24Hour={false}
  isDisabled={false}
  menuAppendTo="inline"
  placeholder="hh:mm"
  stepMinutes={30}
  time=""
  validateTime={[Function]}
  value="00:00"
  width={150}
>
  <div
    className="pf-c-date-picker"
  >
    <div
      className="pf-c-date-picker__input"
      style={
        Object {
          "--pf-c-date-picker__input--c-form-control--Width": 150,
        }
      }
      value="00:00"
    >
      <InputGroup>
        <div
          className="pf-c-input-group"
        >
          <div
            className="pf-c-select"
            id="time-picker-16252543453491o6bvfvutjx"
            key=".0"
          >
            <div
              className="pf-c-select__toggle pf-m-typeahead"
              style={
                Object {
                  "paddingLeft": "0",
                }
              }
            >
              <TextInput
                aria-label="time picker"
                autoComplete="off"
                className="pf-c-form-control pf-c-select__toggle-typeahead"
                iconVariant="clock"
                id="time-picker-16252543453491o6bvfvutjx-input"
                isDisabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                placeholder="hh:mm"
                type="text"
                validated="error"
                value="01:00"
              >
                <TextInputBase
                  aria-label="time picker"
                  autoComplete="off"
                  className="pf-c-form-control pf-c-select__toggle-typeahead"
                  iconVariant="clock"
                  id="time-picker-16252543453491o6bvfvutjx-input"
                  innerRef={
                    Object {
                      "current": <input
                        aria-invalid="true"
                        aria-label="time picker"
                        autocomplete="off"
                        class="pf-c-form-control pf-m-icon pf-m-clock pf-c-form-control pf-c-select__toggle-typeahead"
                        id="time-picker-16252543453491o6bvfvutjx-input"
                        placeholder="hh:mm"
                        type="text"
                        value="01:00"
                      />,
                    }
                  }
                  isDisabled={false}
                  isLeftTruncated={false}
                  isReadOnly={false}
                  isRequired={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  placeholder="hh:mm"
                  type="text"
                  validated="error"
                  value="01:00"
                >
                  <input
                    aria-invalid={true}
                    aria-label="time picker"
                    autoComplete="off"
                    className="pf-c-form-control pf-m-icon pf-m-clock pf-c-form-control pf-c-select__toggle-typeahead"
                    disabled={false}
                    id="time-picker-16252543453491o6bvfvutjx-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    placeholder="hh:mm"
                    readOnly={false}
                    required={false}
                    type="text"
                    value="01:00"
                  />
                </TextInputBase>
              </TextInput>
            </div>
          </div>
        </div>
      </InputGroup>
      <div
        className="pf-c-date-picker__helper-text pf-m-error"
      >
        Invalid time format
      </div>
    </div>
  </div>
</TimePicker>
`;

exports[`test isInvalid should be valid by default 1`] = `
<TimePicker
  aria-label="time picker"
  className=""
  delimiter=":"
  direction="down"
  inputProps={Object {}}
  invalidFormatErrorMessage="Invalid time format"
  is24Hour={false}
  isDisabled={false}
  menuAppendTo="inline"
  placeholder="hh:mm"
  stepMinutes={30}
  time=""
  validateTime={[Function]}
  value="00:00"
  width={150}
>
  <div
    className="pf-c-date-picker"
  >
    <div
      className="pf-c-date-picker__input"
      style={
        Object {
          "--pf-c-date-picker__input--c-form-control--Width": 150,
        }
      }
      value="00:00"
    >
      <InputGroup>
        <div
          className="pf-c-input-group"
        >
          <div
            className="pf-c-select"
            id="time-picker-1625254345331ey5hew7j51"
            key=".0"
          >
            <div
              className="pf-c-select__toggle pf-m-typeahead"
              style={
                Object {
                  "paddingLeft": "0",
                }
              }
            >
              <TextInput
                aria-label="time picker"
                autoComplete="off"
                className="pf-c-form-control pf-c-select__toggle-typeahead"
                iconVariant="clock"
                id="time-picker-1625254345331ey5hew7j51-input"
                isDisabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                placeholder="hh:mm"
                type="text"
                validated="default"
                value=""
              >
                <TextInputBase
                  aria-label="time picker"
                  autoComplete="off"
                  className="pf-c-form-control pf-c-select__toggle-typeahead"
                  iconVariant="clock"
                  id="time-picker-1625254345331ey5hew7j51-input"
                  innerRef={
                    Object {
                      "current": <input
                        aria-invalid="false"
                        aria-label="time picker"
                        autocomplete="off"
                        class="pf-c-form-control pf-m-icon pf-m-clock pf-c-form-control pf-c-select__toggle-typeahead"
                        id="time-picker-1625254345331ey5hew7j51-input"
                        placeholder="hh:mm"
                        type="text"
                        value=""
                      />,
                    }
                  }
                  isDisabled={false}
                  isLeftTruncated={false}
                  isReadOnly={false}
                  isRequired={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  placeholder="hh:mm"
                  type="text"
                  validated="default"
                  value=""
                >
                  <input
                    aria-invalid={false}
                    aria-label="time picker"
                    autoComplete="off"
                    className="pf-c-form-control pf-m-icon pf-m-clock pf-c-form-control pf-c-select__toggle-typeahead"
                    disabled={false}
                    id="time-picker-1625254345331ey5hew7j51-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    placeholder="hh:mm"
                    readOnly={false}
                    required={false}
                    type="text"
                    value=""
                  />
                </TextInputBase>
              </TextInput>
            </div>
          </div>
        </div>
      </InputGroup>
    </div>
  </div>
</TimePicker>
`;

exports[`test isInvalid should stay valid after onChange 1`] = `
<TimePicker
  aria-label="time picker"
  className=""
  delimiter=":"
  direction="down"
  inputProps={Object {}}
  invalidFormatErrorMessage="Invalid time format"
  is24Hour={false}
  isDisabled={false}
  menuAppendTo="inline"
  placeholder="hh:mm"
  stepMinutes={30}
  time=""
  validateTime={[Function]}
  value="00:00"
  width={150}
>
  <div
    className="pf-c-date-picker"
  >
    <div
      className="pf-c-date-picker__input"
      style={
        Object {
          "--pf-c-date-picker__input--c-form-control--Width": 150,
        }
      }
      value="00:00"
    >
      <InputGroup>
        <div
          className="pf-c-input-group"
        >
          <div
            className="pf-c-select"
            id="time-picker-16252543453415rwog39tiiu"
            key=".0"
          >
            <div
              className="pf-c-select__toggle pf-m-typeahead"
              style={
                Object {
                  "paddingLeft": "0",
                }
              }
            >
              <TextInput
                aria-label="time picker"
                autoComplete="off"
                className="pf-c-form-control pf-c-select__toggle-typeahead"
                iconVariant="clock"
                id="time-picker-16252543453415rwog39tiiu-input"
                isDisabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                placeholder="hh:mm"
                type="text"
                validated="default"
                value="01:00"
              >
                <TextInputBase
                  aria-label="time picker"
                  autoComplete="off"
                  className="pf-c-form-control pf-c-select__toggle-typeahead"
                  iconVariant="clock"
                  id="time-picker-16252543453415rwog39tiiu-input"
                  innerRef={
                    Object {
                      "current": <input
                        aria-invalid="false"
                        aria-label="time picker"
                        autocomplete="off"
                        class="pf-c-form-control pf-m-icon pf-m-clock pf-c-form-control pf-c-select__toggle-typeahead"
                        id="time-picker-16252543453415rwog39tiiu-input"
                        placeholder="hh:mm"
                        type="text"
                        value="01:00"
                      />,
                    }
                  }
                  isDisabled={false}
                  isLeftTruncated={false}
                  isReadOnly={false}
                  isRequired={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  placeholder="hh:mm"
                  type="text"
                  validated="default"
                  value="01:00"
                >
                  <input
                    aria-invalid={false}
                    aria-label="time picker"
                    autoComplete="off"
                    className="pf-c-form-control pf-m-icon pf-m-clock pf-c-form-control pf-c-select__toggle-typeahead"
                    disabled={false}
                    id="time-picker-16252543453415rwog39tiiu-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    placeholder="hh:mm"
                    readOnly={false}
                    required={false}
                    type="text"
                    value="01:00"
                  />
                </TextInputBase>
              </TextInput>
            </div>
          </div>
        </div>
      </InputGroup>
    </div>
  </div>
</TimePicker>
`;
